src/main/java/com/autotune/common/utils/CommonUtils.java

```diff
@@ -315,4 +335,164 @@ public static boolean isInvalidDataSource(DataSourceInfo datasource) {
+                                                               Map<String, Terms> termsMap,
+                                                               Double measurementDurationMinutesInDouble,
+                                                               String gpuDetectionQuery)
+            throws IOException, NoSuchAlgorithmException, KeyStoreException,
+            KeyManagementException, ParseException, FetchMetricsError {
+
+        SimpleDateFormat sdf = new SimpleDateFormat(KruizeConstants.DateFormats.STANDARD_JSON_DATE_FORMAT, Locale.ROOT);
+        String containerName = containerData.getContainer_name();
+        String queryToEncode = null;
+        long interval_end_time_epoc = 0;
+        long interval_start_time_epoc = 0;
+
+        LOGGER.info("maxDateQuery: {}", maxDateQuery);
+        queryToEncode =  maxDateQuery
+                .replace(AnalyzerConstants.NAMESPACE_VARIABLE, namespace)
+                .replace(AnalyzerConstants.CONTAINER_VARIABLE, containerName)
+                .replace(AnalyzerConstants.WORKLOAD_VARIABLE, workload)
+                .replace(AnalyzerConstants.WORKLOAD_TYPE_VARIABLE, workload_type);
+
+        String dateMetricsUrl = String.format(KruizeConstants.DataSourceConstants.DATE_ENDPOINT_WITH_QUERY,
+                dataSourceInfo.getUrl(),
+                URLEncoder.encode(queryToEncode, CHARACTER_ENCODING)
+        );
+
+        LOGGER.info(dateMetricsUrl);
+        GenericRestApiClient client = new GenericRestApiClient(dataSourceInfo);
+        client.setBaseURL(dateMetricsUrl);
+        JSONObject genericJsonObject = client.fetchMetricsJson(KruizeConstants.APIMessages.GET, "");
+        JsonObject jsonObject = new Gson().fromJson(genericJsonObject.toString(), JsonObject.class);
+        JsonArray resultArray = jsonObject.getAsJsonObject(KruizeConstants.JSONKeys.DATA).getAsJsonArray(KruizeConstants.DataSourceConstants.DataSourceQueryJSONKeys.RESULT);
+
+        if (null == resultArray || resultArray.isEmpty()) {
+            // Need to alert that container max duration is not detected
+            // Ignoring it here, as we take care of it at generate recommendations
+            return;
+        }
+
+        resultArray = resultArray.get(0)
+                .getAsJsonObject().getAsJsonArray(KruizeConstants.DataSourceConstants.DataSourceQueryJSONKeys.VALUE);
+        long epochTime = resultArray.get(0).getAsLong();
+        String timestamp = sdf.format(new Date(epochTime * KruizeConstants.TimeConv.NO_OF_MSECS_IN_SEC));
+        Date date = sdf.parse(timestamp);
+        Timestamp dateTS = new Timestamp(date.getTime());
+        interval_end_time_epoc = dateTS.getTime() / KruizeConstants.TimeConv.NO_OF_MSECS_IN_SEC
+                - ((long) dateTS.getTimezoneOffset() * KruizeConstants.TimeConv.NO_OF_SECONDS_PER_MINUTE);
+        int maxDay = Terms.getMaxDays(termsMap);
+        LOGGER.info(KruizeConstants.APIMessages.MAX_DAY, maxDay);
```

Comment:
Please evaluate if we need all of these LOGGER's and if it is really needed, add them as LOGGER.debug